### ISSUE TITLE
VLS PGM Coefficients with different angle of incidence

### ISSUE THREAD

**Issue by Telschazz** [oasys-kit/ShadowOui#223]
Hi,

I'm using the "VLS PGM Coefficients Calculator" and like it very much.
However my PGM-optic has a pitch-offset on the plane mirror so that the beam enters horizontally but exits the grating in a slight upward angle.
I wanted to insert this into my OASYS setup so I tried editing the mirror by hand after the "VLS PGM Coefficients Calculator" set it's values and indeed it worked as I expected. Now I wanted to do this automatically and inserted a Python script, which would add my offset to the alpha- and beta-angles of the "PreProcessor_Data" from the calculator. 

![OASYS VLS](https://user-images.githubusercontent.com/11213750/132493600-cd763fb7-df4c-4817-b399-599158baff2c.png)

Unfortunately this seems to affect the global setup of the "VLS-PGM PreProcessor Data" and modifies the angles of the grating as well.
How can I do this?

Regards
Jörn

**Comment by srio** [comment]
Could you send your script? In principle you should duplicate the variables, do not modify the variables that you receive in in_object_1. 
Regards, Manuel

**Comment by Telschazz** [comment]
I'm using:

```
dat = in_object_1
dat.alpha = dat.alpha + 0.6
dat.beta = dat.beta + 0.6
out_object = dat

```

**Comment by Telschazz** [comment]
It modifies the angles even when I'm building it like this

![OASYS VLS1](https://user-images.githubusercontent.com/11213750/132678255-acc21d6f-5c26-428a-96f6-aba69bce635a.png)

**Comment by srio** [comment]
Indeed, your in_object_1, dat, and out_object are the same. It does not matter what you change, everything is changed.  

Try this:
```
import copy
out_object = copy.deepcopy(in_object_1)
out_object.alpha = out_object.alpha + 0.6
out_object.beta = out_object.beta + 0.6
```

**Comment by Telschazz** [comment]
Thanks a lot!
I just didn't realize what you meant.
I thought I knew my Python ;-)
